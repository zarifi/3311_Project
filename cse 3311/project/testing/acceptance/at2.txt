-- at2.txt: Normal prescription followed by dangerous one
add_physician	   (1, "Hippocrates", generalist)
add_physician	   (-3, "Mayo", generalist)
-- negative id
add_physician	   (2, "Galen", specialist) 
--  wrong type
add_physician      (2,"2Sara", generalist) 
-- name start with number duplicate id

add_physician    (5,"Hossein", specialist)
add_physician 	   (3, "Gayl", generalist)
--add_physician     ("Gorge", 4, generalist) 
-- arguments not in order 
add_physician	   (2, "Galen", specialist)
add_physician	   (10,"",specialist) 
-- empty name
add_physician	   (11.0, "Leo",generalist) 
-- not int
add_physician	   (11, "$haily", generalist) 
-- name start with nonletter  

add_patient        (-3, "Dora") 
-- negative id
add_patient        (1, "&Phineas") 
-- start with non Letter
add_patient        (1, "Henry") 
add_patient		   (5, "phil")
add_patient		   (2, "Joe")
add_patient		   (2,"Jane") 
-- already in use id
add_patient 	   (6, "phil") 
-- already in use names
add_patient 	   (3, "saeed")
--add_patient	   (99, "Ebi", "sia") 
-- wrong number of arguments
add_patient 	   (33, "") 
-- empty name
add_patient	       (3, "adel") 
-- not int
--add_patient 	   (3, "harry", "henry") 
-- wrong number of arguments


-- physicians and patients are displayed in id order


--add_medication   (id: ID; name: NAME; kind: MEDICATION_TYPE; low: VALUE; hi: VALUE)
add_medication     (1, ["Wafarin", pill, 1.0, 6.0])
add_medication     (2, ["Sulfamethizole", pill, 200.0, 500.0]) 
add_medication     (4, ["m4", liquid, 2.5, 9.5])
-- medication with name m4 is already in
add_medication     (5, ["m4", liquid, 10.0, 25.5])
add_medication     (5, ["m5", liquid, 7.0, 16.0])
add_medication     (3, ["m3", liquid, 10.0, 25.5])

add_interaction    (1,2)
-- the following will not add
add_interaction    (2,1)
add_interaction    (1,2)
add_interaction    (5,3)
--interactions are displayed alphabetically by name

-- prescription id 1: generalist Galen prescribes for Dora
new_prescription   (1, 2, 3)
dpr_q

-- add Wafarin 5.5 mg to prescription id 1
add_medicine       (1, 1, 5.5)
-- add Sulfamethizole 300.5 mg to prescription id 1
add_medicine       (1, 2, 300.5)
  -- Hippocrates is a generalist and this conflicts

add_medicine       (1, 3, 12.5)
dpr_q

-- prescription id 3: Mayo prescribes for Phineas
new_prescription   (3, 2, 1)
-- add Sulfamethizole 310.5 mg to prescription id 3
add_medicine       (3, 2, 300.5)

add_medicine       (3, 3, 12.5)

	-- Added by Simon
add_medicine       (3, 5, 12.5)


-- query which patients are taking Sulfamethizole?
prescriptions_q(2)

-- there should be a dpr, dangerous prescription report
dpr_q

remove_medicine(3,5)

dpr_q
