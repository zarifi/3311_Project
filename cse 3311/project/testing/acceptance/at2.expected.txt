  0: ok
  Physicians:
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(1,"Hippocrates",generalist)
  1: ok
  Physicians:
    1->[Hippocrates,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(-3,"Mayo",generalist)
  2: physician id must be a positive integer
  Physicians:
    1->[Hippocrates,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(2,"Galen",specialist)
  3: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(2,"2Sara",generalist)
  4: physician id already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(5,"Hossein",specialist)
  5: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    5->[Hossein,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(3,"Gayl",generalist)
  6: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(2,"Galen",specialist)
  7: physician id already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(10,"",specialist)
  8: name must start with a letter
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(11,"Leo",generalist)
  9: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(11,"$haily",generalist)
  10: physician id already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(-3,"Dora")
  11: patient id must be a positive integer
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(1,"&Phineas")
  12: name must start with a letter
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(1,"Henry")
  13: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(5,"phil")
  14: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    5->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(2,"Joe")
  15: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    5->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(2,"Jane")
  16: patient id already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    5->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(6,"phil")
  17: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    5->phil
    6->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(3,"saeed")
  18: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(33,"")
  19: name must start with a letter
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(3,"adel")
  20: patient id already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
  Interactions: 
  Prescriptions:
->add_medication(1,["Wafarin", pill, 1, 6])
  21: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
  Interactions: 
  Prescriptions:
->add_medication(2,["Sulfamethizole", pill, 200, 500])
  22: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
  Interactions: 
  Prescriptions:
->add_medication(4,["m4", liquid, 2.5, 9.5])
  23: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    4->[m4,lq,2.50,9.50]
  Interactions: 
  Prescriptions:
->add_medication(5,["m4", liquid, 10, 25.5])
  24: medication name already in use
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    4->[m4,lq,2.50,9.50]
  Interactions: 
  Prescriptions:
->add_medication(5,["m5", liquid, 7, 16])
  25: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
  Prescriptions:
->add_medication(3,["m3", liquid, 10, 25.5])
  26: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
  Prescriptions:
->add_interaction(1,2)
  27: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
  Prescriptions:
->add_interaction(2,1)
  28: interaction already exists
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
  Prescriptions:
->add_interaction(1,2)
  29: interaction already exists
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
  Prescriptions:
->add_interaction(5,3)
  30: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
->new_prescription(1,2,3)
  31: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,()]
->dpr_q
  32: ok
  There are no dangerous prescriptions
->add_medicine(1,1,5.5)
  33: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50)]
->add_medicine(1,2,300.5)
  34: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50)]
->add_medicine(1,3,12.5)
  35: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
->dpr_q
  36: ok
  There are dangerous prescriptions:
    (saeed,3)->{ [Sulfamethizole,pl,2]->[Wafarin,pl,1] }
->new_prescription(3,2,1)
  37: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
    3->[3,2,1,()]
->add_medicine(3,2,300.5)
  38: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
    3->[3,2,1,(2->300.50)]
->add_medicine(3,3,12.5)
  39: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
    3->[3,2,1,(2->300.50,3->12.50)]
->add_medicine(3,5,12.5)
  40: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
    3->[3,2,1,(2->300.50,3->12.50,5->12.50)]
->prescriptions_q(2)
  41: ok
  Output: medication is Sulfamethizole
    1->Henry
    3->saeed
->dpr_q
  42: ok
  There are dangerous prescriptions:
    (Henry,1)->{ [m3,lq,3]->[m5,lq,5] }
    (saeed,3)->{ [Sulfamethizole,pl,2]->[Wafarin,pl,1] }
->remove_medicine(3,5)
  43: ok
  Physicians:
    1->[Hippocrates,gn]
    2->[Galen,sp]
    3->[Gayl,gn]
    5->[Hossein,sp]
    11->[Leo,gn]
  Patients: 
    1->Henry
    2->Joe
    3->saeed
    5->phil
    6->phil
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[Sulfamethizole,pl,200.00,500.00]
    3->[m3,lq,10.00,25.50]
    4->[m4,lq,2.50,9.50]
    5->[m5,lq,7.00,16.00]
  Interactions: 
    [Sulfamethizole,pl,2]->[Wafarin,pl,1]
    [m3,lq,3]->[m5,lq,5]
  Prescriptions:
    1->[1,2,3,(1->5.50,2->300.50,3->12.50)]
    3->[3,2,1,(2->300.50,3->12.50)]
->dpr_q
  44: ok
  There are dangerous prescriptions:
    (saeed,3)->{ [Sulfamethizole,pl,2]->[Wafarin,pl,1] }
